### PR TITLE
backup foreman

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: puppet/modules/profiles/manifests/foreman.pp [hunk 1]
```diff
@@ -11,4 +11,26 @@
...
+  file { $backup_path:
+    ensure => directory,
+  }
+
+  include profiles::backup::sender
+
+  restic::repository { 'foreman':
+    backup_cap_dac_read_search => true,
+    backup_path                => $backup_path,
+    backup_pre_cmd             => ["+/usr/bin/foreman-maintain backup online --assumeyes --preserve-directory ${backup_path}"],
```
Comment (evgeni) [inline puppet/modules/profiles/manifests/foreman.pp:34]: https://www.freedesktop.org/software/systemd/man/systemd.service.html#Command%20lines is neat

Comment (ekohl) [reply puppet/modules/profiles/manifests/foreman.pp:34]: I did not know systemd could do this, but nice!

File: puppet/modules/profiles/manifests/foreman.pp [hunk 2]
```diff
@@ -11,4 +11,26 @@
...
+    ensure => present,
+  }
+
+  $backup_base_path = '/var/backups'
+  $backup_path = "${backup_base_path}/foreman"
+
+  file { $backup_base_path:
+    ensure => directory,
+  }
+  file { $backup_path:
```
Comment (ekohl) [inline puppet/modules/profiles/manifests/foreman.pp:25]: You can shorten it:
```suggestion
  file { [$backup_base_path, $backup_path]:
    ensure => directory,
  }
```

**Review by ekohl — APPROVED** [review]

File: puppet/modules/profiles/manifests/foreman.pp (hunk 1 above)
Comment (evgeni) [reply puppet/modules/profiles/manifests/foreman.pp:34]: Neither did I until a few days ago :)

**Comment by evgeni** [comment]
FTR, I had to apply the following patches locally to get things to work:
- https://github.com/theforeman/foreman_maintain/pull/766
- https://github.com/theforeman/foreman_maintain/pull/771

**Pull request merged**